### PR TITLE
[DEV-4443] Switch submission fields for downloads

### PR DESCRIPTION
**Pull request by emily-brents**
**Description:**
Leftovers from DEV-4443, essentially just removes the `last_reported_submission_period` and replaces it with `submission_date` for the Custom Account Downloads downloads.

**Technical details:**
Adds `submission_date` to file A and B download, alters the `submission_date` for file C to allow for period or quarter formatting.

**Requirements for PR merge:**

1. [X] Unit & integration tests updated
2. [X] API documentation updated
3. [X] Necessary PR reviewers:
    - [x] Backend
4. [N/A] Matview impact assessment completed
5. [N/A] Frontend impact assessment completed
6. [X] Data validation completed
7. [N/A] Appropriate Operations ticket(s) created
8. [ ] Jira Ticket [DEV-4443](https://federal-spending-transparency.atlassian.net/browse/DEV-4443):
    - [ ] Link to this Pull-Request
    - [ ] Performance evaluation of affected (API | Script | Download)
    - [ ] Before / After data comparison

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

File: usaspending_api/download/v2/download_column_historical_lookups.py [hunk 1]
```diff
@@ -1567,11 +1561,8 @@
             [
                 ("owning_agency_name", "treasury_account__funding_toptier_agency__name"),
                 ("reporting_agency_name", "submission__reporting_agency_name"),
-                (
-                    "last_reported_submission_period",
-                    "last_reported_submission_period",
-                ),  # Column is annotated in account_download.py
-                ("allocation_transfer_agency_identifier_code", "treasury_account__allocation_transfer_agency_id"),
+                ("submission_period", "submission_period",),  # Column is annotated in account_download.py
+                ("allocation_transfer_agency_identifer_code", "treasury_account__allocation_transfer_agency_id"),
```
Comment (tony-sappe) [inline usaspending_api/download/v2/download_column_historical_lookups.py:1565]: Was this intentional?
```suggestion
                ("allocation_transfer_agency_identifier_code", "treasury_account__allocation_transfer_agency_id"),
```

File: usaspending_api/common/helpers/orm_helpers.py [hunk 2]
```diff
@@ -71,6 +71,17 @@ class FiscalYearAndQuarter(Func):
     output_field = CharField()
 
 
+class FiscalYearAndPeriod(Func):
```
Comment (kbard) [inline usaspending_api/common/helpers/orm_helpers.py:74]: I believe this function and the annotation to support it is a duplicate of `get_fyp_or_q_notation` four functions below.

File: usaspending_api/download/v2/download_column_historical_lookups.py (hunk 1 above)
Comment (kbard) [reply usaspending_api/download/v2/download_column_historical_lookups.py:1565]: OMG PLEASE DON'T LET THIS SPELLING MISTAKE BACK INTO THE CODE BASE!

File: usaspending_api/common/helpers/orm_helpers.py (hunk 2 above)
Comment (emily-brents) [reply usaspending_api/common/helpers/orm_helpers.py]: it is! wish i could read. remove function and replaced w/ `get_fyp_or_q_notation`

**Pull request merged**